omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -587,10 +582,27 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
                     self._table.numberOfPages = Math.floor(initialRowCount/self._table.fnSettings()._iDisplayLength)+1;
 
             } else if(self.options.searchPhrase || self.options.doSearchWhenEmpty) {
-                if (self.options.searchPhrase == null) {
-                    self.options.searchPhrase = "";
-                }
-                $j(input).val(self.options.searchPhrase).keyup();
+            	if (self.options.searchPhrase == null) {
+            		self.options.searchPhrase = "";
+            	}                
+            	//check to see if we need to select the checkbox for includeVerbose and/or includeVoided
+            	//since there is a searchPhrase the page will reload using these values
+            	if(self.options.showVerbose){
+            		$j('#includeVerbose').attr('checked','checked');
+            	}
+            	if(self.options.includeVoided){
+            		$j('#includeVoided').attr('checked','checked');
+            	}
+            	$j(input).val(self.options.searchPhrase).keyup();
+            	//setting the cursor to the end of the searchPhrase
+            	//setSelectionRange is a method of the DOM element not the jQuery object so get the DOM element
+            	var inputNodeDOM = $j('#inputNode').get(0);
+            	if(inputNodeDOM){
+            		var searchPhraseLength = self.options.searchPhrase.length;
+            		if (inputNodeDOM.setSelectionRange) {
+            			inputNodeDOM.setSelectionRange(searchPhraseLength, searchPhraseLength);
```

Comment:
I guess I was misunderstanding your previous comment. What are you not sure we need from previous commits? I thought you were refering to the whole section here. I redid this section trying to make sure we had a search loaded when the checkboxes were checked and I guess in my logic I was only doing the keyup when we had a searchPhrase. I can move the keyup and cursor code outside of the else if. I should have double checked the original functionality and realized it was meant to work in other scenarious too because it will still trigger it when I need it for the checkboxes too.